elasticjob-infra/elasticjob-registry-center/elasticjob-regitry-center-provider/elasticjob-registry-center-zookeeper-curator/src/main/java/org/apache/shardingsphere/elasticjob/reg/zookeeper/ZookeeperRegistryCenter.java

```diff
@@ -408,20 +409,28 @@ public void executeInLeader(final String key, final LeaderExecutionCallback call
     }
     
     @Override
-    public void watch(final String key, final DataChangedEventListener listener) {
+    public void watch(final String key, final DataChangedEventListener listener, final Executor executor) {
         CuratorCache cache = caches.get(key + "/");
-        cache.listenable().addListener((curatorType, oldData, newData) -> {
-            if (null == newData && null == oldData) {
-                return;
+        CuratorCacheListener cacheListener = new CuratorCacheListener() {
```

Comment:
Why change this to anonymous class?

Reply:
> Why change this to anonymous class?

I'm not familiar with lamda expression,  so I use anonymous class to resuse the code. I can change it back to lamda expression.